host/traceh.c

```diff
@@ -310,23 +314,14 @@ oe_result_t oe_log(oe_log_level_t level, const char* fmt, ...)
 void oe_log_message(bool is_enclave, oe_log_level_t level, const char* message)
 {
     // get timestamp for log
-#if defined(__linux__)
-    struct timeval time_now;
-    gettimeofday(&time_now, NULL);
-    struct tm* t = gmtime(&time_now.tv_sec);
-#else
+    struct tm t;
     time_t lt = time(NULL);
-    struct tm* t = gmtime(&lt);
-#endif
+    gmtime_r(&lt, &t);
```

Comment:
Doesn't gmtime_r not work on Windows? Or is this function only ever called from the enclave?

Reply:
I created a macro in a header file so that it would change it to gmtime_s on Windows.

Reply:
I would prefer a wrapper function instead (like oe_gmtime_r) and do an ifdef there. I think a macros hides too much of what is actually going on

Reply:
This is a matter of taste, but I have the opposite preference from @jhand2 for anything that's part of the standard.  https://en.cppreference.com/w/c/chrono/gmtime shows that gmtime_r is part of the C11 standard.

Reply:
Ah, didn't realize gmtime_r was a part of the new standard (not C11 though, it shows as C2x). I don't have a strong preference either way then

Reply:
Resolving this, I think it's fine to leave as-is